sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c

```diff
@@ -841,7 +843,7 @@ ssize_t gnrc_tcp_recv(gnrc_tcp_tcb_t *tcb, void *data, const size_t max_len,
     /* Setup connection timeout */
     _sched_connection_timeout(&tcb->event_misc, &mbox);
 
-    if (timeout_duration_ms > 0) {
+    if (timeout_duration_ms != GNRC_TCP_NO_TIMEOUT) {
```

Comment:
Not

```suggestion
    if ((timeout_duration_ms > 0) && (timeout_duration_ms != GNRC_TCP_NO_TIMEOUT)) {
```

as above?

Reply:
Nope. I noticed that timeout_duration_ms was checked before leading to an early return so this comparison is always true and can be removed.